### ISSUE TITLE
About the labels of IEMOCAP

### ISSUE THREAD

**Issue by yuki9965** [Justin1904/Low-rank-Multimodal-Fusion#11]
hi:

  First of all thank you very much for your work. But I don't quite understand some processing about iemocap dataset. 

 - First, iemocap.pkl is divided into four parts according to emotion. The label of each part is a 2-D array with 0 or 1.  What this label refers to？And IEMOCAP should be a four-category multi-label classification task. Why divide the data set like this ?

- Second, I don't quite understand the code in train_iemocap.py : 
      `torch.max(y, 1)[1]` . 
This line of code also confuses me because I don’t understand the meaning of the label.

I will be very grateful If you can help me.

**Comment by Justin1904** [comment]
The first question - indeed IEMOCAP is a 4-category multi-label classification in nature. It's just that during our experiment we separately trained models for each emotion, treating it as 4 binary classification problems. The 2-d labels are something like [1, 0] for positive and [0, 1] for negative.

Second question - it's taking the argmax of the label so that [1, 0] -> 0, [0, 1] -> 1. The cross-entropy loss takes class id as the label, not one-hot vectors, that's why we do this conversion.

**Comment by yuki9965** [comment]
> The first question - indeed IEMOCAP is a 4-category multi-label classification in nature. It's just that during our experiment we separately trained models for each emotion, treating it as 4 binary classification problems. The 2-d labels are something like [1, 0] for positive and [0, 1] for negative.
> 
> Second question - it's taking the argmax of the label so that [1, 0] -> 0, [0, 1] -> 1. The cross-entropy loss takes class id as the label, not one-hot vectors, that's why we do this conversion.

thanks for your reply, I figure out now.

**Comment by Grace7-leader** [comment]
Hello, I download your code to the local implementation, but train_ mosi.py There is a gradient explosion in it. I only changed the path. The environment is Python 1.5.1

**Comment by Justin1904** [comment]
@Grace7-leader can you: 1) identify if you are seeing the same issue as #10 ; 2) if 1) is not the case, open another issue to keep track?

**Comment by Grace7-leader** [comment]
I have solved this problem.but the NAN value keeps appearing in train.mosi.py, and I have adopted the gradient clipping method and still cannot completely solve the NAN problem.




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Zhun"<notifications@github.com&gt;; 
发送时间: 2020年9月5日(星期六) 下午5:51
收件人: "Justin1904/Low-rank-Multimodal-Fusion"<Low-rank-Multimodal-Fusion@noreply.github.com&gt;; 
抄送: "   邮件通知"<708949606@qq.com&gt;; "Mention"<mention@noreply.github.com&gt;; 
主题: Re: [Justin1904/Low-rank-Multimodal-Fusion] About the labels of IEMOCAP (#11)





 
@Grace7-leader can you: 1) identify if you are seeing the same issue as #10 ; 2) if 1) is not the case, open another issue to keep track?
